contracts/CollateralFundV1.sol

```diff
@@ -2,131 +2,155 @@
+    //slither-disable-next-line naming-convention
+    constructor(address _usdcAddress, address _hubAddress) {
+        owner = msg.sender;
+        usdcAddress = _usdcAddress;
+        usdcContract = IERC20(_usdcAddress);
         hubAddress = _hubAddress;
     }
 
+    /**************************************** Fund / Withdraw ****************************************/
+
     /**
-     * @notice Send collateral to the fund.
+     * @notice Deposit collateral to the fund.
+     * @param _amount the amount of USDC token send to the fund.
      */
-    function fund() external payable {
-        // TODO: change to USDC instead of Ether
-        collateralFundedByAddress[msg.sender] += msg.value;
-        totalCollateral += msg.value;
-        funders.push(msg.sender);
-        emit Deposit(
-            msg.sender,
-            msg.value,
-            collateralFundedByAddress[msg.sender]
-        );
+    //slither-disable-next-line naming-conventions
+    function deposit(uint256 _amount) external {
+        require(_amount > 0, "Amount should be greator than zero");
+
+        // Send the USDC from the user's wallet to the fund and update the user collateral amount.
+        usdcContract.transferFrom(msg.sender, address(this), _amount);
+        addressToCollateralAmount[msg.sender] += _amount;
+        emit Deposit(msg.sender, _amount, addressToCollateralAmount[msg.sender]);
+
+        // Add the sender to the funders list if he's not already in the list.
+        if (!addressToFunderStatus[msg.sender]) {
+            addressToFunderStatus[msg.sender] = true;
+            funders.push(msg.sender);
+            emit NewFunder(msg.sender);
+        }
     }
 
     /**
      * @notice Withdraw collateral from the fund.
      * @param _amount the amount to withdraw from the fund.
      */
-    function withdraw(uint256 _amount) external payable {
-        // TODO: change to USDC instead of Ether
-        require(
-            _amount <= collateralFundedByAddress[msg.sender],
-            "You can't withdraw more than what you deposited"
-        );
-        collateralFundedByAddress[msg.sender] -= _amount;
-        totalCollateral -= _amount;
-        emit Withdraw(
-            msg.sender,
-            _amount,
-            collateralFundedByAddress[msg.sender]
-        );
-        payable(msg.sender).transfer(_amount);
+    //slither-disable-next-line naming-convention
+    function withdraw(uint256 _amount) external {
+        require(_amount > 0, "Amount should be greator than zero");
+        require(addressToFunderStatus[msg.sender], "The user has not deposited any collateral to the fund");
+        require(_amount <= addressToCollateralAmount[msg.sender], "The user cannot withdraw more than what he deposited");
+
+        // Send the USDC from the fund to the user's wallet and update the user collateral amount.
+        usdcContract.transfer(msg.sender, _amount);
+        addressToCollateralAmount[msg.sender] -= _amount;
```

Comment:
update `addressToFunderStatus` if the balance is at 0 again

Reply:
Yeah good catch, i'll add this :)